packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ScmMethodProviderCollection.cs

```diff
@@ -121,6 +121,22 @@ private IEnumerable<MethodBodyStatement> GetStackVariablesForProtocolParamConver
                         statements.Add(UsingDeclare("content", BinaryContentHelperSnippets.FromEnumerable(parameter), out var content));
                         declarations["content"] = content;
                     }
+                    else if (parameter.Type.IsFrameworkType && !parameter.Type.Equals(typeof(BinaryData)))
+                    {
+                        if (parameter.Type.Equals(typeof(string)))
+                        {
+                            var bdExpression = Operation.RequestMediaTypes?.Contains("application/json") == true
+                                ? BinaryDataSnippets.FromObjectAsJson(parameter)
+                                : BinaryDataSnippets.FromString(parameter);
+                            statements.Add(UsingDeclare("content", BinaryContentSnippets.Create(bdExpression), out var content));
+                            declarations["content"] = content;
+                        }
+                        else
```

Comment:
If I understand correctly, this else will be the default behavior, which is not `BinaryData`.
Why are we not dealing with `BinaryData`?

Reply:
binary data does not require any special stack variables to do the conversion so we don't need to deal with it.